src/esp/assets/ResourceManager.cpp

```diff
@@ -82,8 +82,9 @@ constexpr char ResourceManager::DEFAULT_LIGHTING_KEY[];
 constexpr char ResourceManager::DEFAULT_MATERIAL_KEY[];
 constexpr char ResourceManager::WHITE_MATERIAL_KEY[];
 constexpr char ResourceManager::PER_VERTEX_OBJECT_ID_MATERIAL_KEY[];
-ResourceManager::ResourceManager()
-    :
+ResourceManager::ResourceManager(
+    metadata::MetadataMediator::ptr& _metadataMediator)
+    : metadataMediator_(_metadataMediator),
```

Comment:
Passing MM to RM so that it has access to required attributesManagers for current active dataset.